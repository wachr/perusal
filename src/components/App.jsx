import { h } from "preact";
import { useState } from "preact/hooks";
import { BrowserRouter, Link } from "react-router-dom";
import { StaticRouter } from "react-router-dom/server";

import baseroute from "../utils/baseroute";
import AppBar from "@mui/material/AppBar";
import Box from "@mui/material/Box";
import ButtonBase from "@mui/material/ButtonBase";
import Stack from "@mui/material/Stack";
import Toolbar from "@mui/material/Toolbar";
import Typography from "@mui/material/Typography";
import { ThemeProvider, createTheme } from "@mui/material/styles";

import StateInspector from "./StateInspector";

const theme = createTheme({
  palette: {
    type: "light",
    primary: {
      main: "#815d56",
      light: "#be9c91",
      dark: "#5f4339",
    },
    secondary: {
      main: "#827717",
      light: "#b4a345",
      dark: "#524a00",
    },
    background: {
      paper: "#e6e6e6",
    },
  },
});

const Router = (props) => {
  if (typeof window !== "undefined") return <BrowserRouter {...props} />;
  return <StaticRouter {...props} />;
};

export default function App() {
  const [appState, setAppState] = useState({});
  return (
    <div id="App">
      <Router basename={baseroute}>
        <ThemeProvider theme={theme}>
          <AppBar>
            <Box>
              <Link to={"/"}>
                <ButtonBase size={"small"}>
                  <Stack>
                    <img
                      src={`${baseroute}/assets/icon.svg`}
                      width={32}
                      height={32}
                    />
                    <Typography variant="caption">perusal</Typography>
                  </Stack>
                </ButtonBase>
              </Link>
            </Box>
          </AppBar>
          <Toolbar />
          <Typography variant="h1">Hello, World!</Typography>
          <StateInspector nodeState={appState} setNode={setAppState} />
        </ThemeProvider>
      </Router>
    </div>
  );
}
