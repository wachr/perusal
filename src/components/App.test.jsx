import { h } from "preact";
import { render, screen } from "@testing-library/preact";

import App from "./App";

describe(App.name, () => {
  it("renders", () => {
    const { container } = render(<App />);
    expect(container).toMatchSnapshot();
  });
});
