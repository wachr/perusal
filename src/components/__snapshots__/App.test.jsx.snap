// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders 1`] = `
<div>
  <div
    id="App"
  >
    <header
      class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionFixed mui-fixed css-hxfn6i-MuiPaper-root-MuiAppBar-root"
    >
      <a
        class="MuiButtonBase-root css-10d1a0h-MuiButtonBase-root"
        href="/"
        tabindex="0"
      >
        <div
          class="css-nen11g-MuiStack-root"
        >
          <img
            height="32"
            src="/assets/icon.svg"
            width="32"
          />
          <span
            class="MuiTypography-root MuiTypography-caption css-1sn4lm3-MuiTypography-root"
          >
            perusal
          </span>
        </div>
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </a>
    </header>
    <div
      class="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular css-1t29gy6-MuiToolbar-root"
    />
    <div
      id="TopicCards"
    >
      <div
        data-testid="topic-card"
        id="TopicCard"
      >
        <div
          class="MuiBox-root css-1s8p6ck"
        >
          <div
            class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-i95gtt-MuiPaper-root-MuiCard-root"
          >
            <div
              class="css-nen11g-MuiStack-root"
            >
              <div
                data-testid="topic-title"
              >
                <a
                  class="MuiTypography-root MuiTypography-h6 MuiLink-root MuiLink-underlineAlways css-103xwh2-MuiTypography-root-MuiLink-root"
                  href="foo"
                >
                  foo
                </a>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        data-testid="topic-card"
        id="TopicCard"
      >
        <div
          class="MuiBox-root css-1s8p6ck"
        >
          <div
            class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-i95gtt-MuiPaper-root-MuiCard-root"
          >
            <div
              class="css-nen11g-MuiStack-root"
            >
              <div
                data-testid="topic-title"
              >
                <a
                  class="MuiTypography-root MuiTypography-h6 MuiLink-root MuiLink-underlineAlways css-103xwh2-MuiTypography-root-MuiLink-root"
                  href="bar"
                >
                  bar
                </a>
              </div>
              <hr
                class="MuiDivider-root MuiDivider-fullWidth css-9mgopn-MuiDivider-root"
              />
              <span
                class="MuiTypography-root MuiTypography-body css-37jdci-MuiTypography-root"
                data-testid="topic-details"
              >
                barbar bar barbarbar
              </span>
            </div>
          </div>
        </div>
      </div>
      <div
        data-testid="topic-card"
        id="TopicCard"
      >
        <div
          class="MuiBox-root css-1s8p6ck"
        >
          <div
            class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-i95gtt-MuiPaper-root-MuiCard-root"
          >
            <div
              data-testid="topic-title"
            >
              <a
                class="MuiTypography-root MuiTypography-h6 MuiLink-root MuiLink-underlineAlways css-103xwh2-MuiTypography-root-MuiLink-root"
                href="baz"
              >
                baz
              </a>
            </div>
          </div>
        </div>
      </div>
      <div
        data-testid="topic-card"
        id="TopicCard"
      >
        <div
          class="MuiBox-root css-1s8p6ck"
        >
          <div
            class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-i95gtt-MuiPaper-root-MuiCard-root"
          >
            <div
              class="css-nen11g-MuiStack-root"
            >
              <div
                data-testid="topic-title"
              >
                <a
                  class="MuiTypography-root MuiTypography-h6 MuiLink-root MuiLink-underlineAlways css-103xwh2-MuiTypography-root-MuiLink-root"
                  href="quux"
                >
                  quux
                </a>
              </div>
              <hr
                class="MuiDivider-root MuiDivider-fullWidth css-9mgopn-MuiDivider-root"
              />
              <ul
                class="MuiList-root MuiList-padding css-h4y409-MuiList-root"
              >
                <li
                  class="MuiListItem-root MuiListItem-gutters css-1wrjpai-MuiListItem-root"
                  data-testid="topic-subtopic"
                >
                  <div
                    class="MuiListItemText-root MuiListItemText-inset css-1fzmbpm-MuiListItemText-root"
                    href="quux/q"
                  >
                    <span
                      class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-10hburv-MuiTypography-root"
                    >
                      q
                    </span>
                  </div>
                </li>
                <li
                  class="MuiListItem-root MuiListItem-gutters css-1wrjpai-MuiListItem-root"
                  data-testid="topic-subtopic"
                >
                  <div
                    class="MuiListItemText-root MuiListItemText-inset css-1fzmbpm-MuiListItemText-root"
                    href="quux/u"
                  >
                    <span
                      class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-10hburv-MuiTypography-root"
                    >
                      u
                    </span>
                  </div>
                </li>
                <li
                  class="MuiListItem-root MuiListItem-gutters css-1wrjpai-MuiListItem-root"
                  data-testid="topic-subtopic"
                >
                  <div
                    class="MuiListItemText-root MuiListItemText-inset css-1fzmbpm-MuiListItemText-root"
                    href="quux/u"
                  >
                    <span
                      class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-10hburv-MuiTypography-root"
                    >
                      u
                    </span>
                  </div>
                </li>
                <li
                  class="MuiListItem-root MuiListItem-gutters css-1wrjpai-MuiListItem-root"
                  data-testid="topic-subtopic"
                >
                  <div
                    class="MuiListItemText-root MuiListItemText-inset css-1fzmbpm-MuiListItemText-root"
                    href="quux/x"
                  >
                    <span
                      class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-10hburv-MuiTypography-root"
                    >
                      x
                    </span>
                  </div>
                </li>
              </ul>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      data-testid="StateInspector-div"
    >
      <div
        class="MuiBox-root css-k0qjc5"
      >
        <div
          class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation0 css-100kxm3-MuiPaper-root"
        >
          <div
            class="MuiBox-root css-109982z"
          >
            <div
              class="MuiBox-root css-d3kwzx"
            >
              <div
                class="MuiButtonGroup-root MuiButtonGroup-outlined css-au5jwi-MuiButtonGroup-root"
                role="group"
              >
                <button
                  class="MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeSmall MuiButton-outlinedSizeSmall MuiButtonBase-root MuiButtonGroup-grouped MuiButtonGroup-groupedHorizontal MuiButtonGroup-groupedOutlined MuiButtonGroup-groupedOutlinedHorizontal MuiButtonGroup-groupedOutlinedPrimary css-nedx0s-MuiButtonBase-root-MuiButton-root"
                  tabindex="0"
                  type="button"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                    data-testid="FileUploadIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M9 16h6v-6h4l-7-7-7 7h4zm-4 2h14v2H5z"
                    />
                  </svg>
                  <span
                    class="MuiTypography-root MuiTypography-button css-1f0on15-MuiTypography-root"
                  >
                    Import State
                  </span>
                  <span
                    class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                  />
                </button>
                <button
                  class="MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeSmall MuiButton-outlinedSizeSmall MuiButtonBase-root MuiButtonGroup-grouped MuiButtonGroup-groupedHorizontal MuiButtonGroup-groupedOutlined MuiButtonGroup-groupedOutlinedHorizontal MuiButtonGroup-groupedOutlinedPrimary css-nedx0s-MuiButtonBase-root-MuiButton-root"
                  tabindex="0"
                  type="button"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                    data-testid="FileDownloadIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M19 9h-4V3H9v6H5l7 7 7-7zM5 18v2h14v-2H5z"
                    />
                  </svg>
                  <span
                    class="MuiTypography-root MuiTypography-button css-1f0on15-MuiTypography-root"
                  >
                    Export State
                  </span>
                  <span
                    class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                  />
                </button>
                <button
                  aria-label="show or hide state"
                  class="MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeSmall MuiButton-outlinedSizeSmall MuiButtonBase-root MuiButtonGroup-grouped MuiButtonGroup-groupedHorizontal MuiButtonGroup-groupedOutlined MuiButtonGroup-groupedOutlinedHorizontal MuiButtonGroup-groupedOutlinedPrimary css-nedx0s-MuiButtonBase-root-MuiButton-root"
                  tabindex="0"
                  type="button"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                    data-testid="ExpandLessIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="m12 8-6 6 1.41 1.41L12 10.83l4.59 4.58L18 14z"
                    />
                  </svg>
                  <span
                    class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                  />
                </button>
              </div>
            </div>
            <div
              aria-label="app state"
              class="MuiFormControl-root MuiFormControl-marginNormal MuiFormControl-fullWidth MuiTextField-root css-17vbkzs-MuiFormControl-root-MuiTextField-root"
            >
              <label
                class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary MuiFormLabel-filled css-1sdofz3-MuiFormLabel-root-MuiInputLabel-root"
                data-shrink="true"
              >
                app state
              </label>
              <div
                class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-multiline css-3fxejv-MuiInputBase-root-MuiOutlinedInput-root"
              >
                <textarea
                  aria-invalid="false"
                  class="MuiOutlinedInput-input MuiInputBase-input MuiInputBase-inputMultiline css-1sqnrkk-MuiInputBase-input-MuiOutlinedInput-input"
                  readonly=""
                  style="height: 0px; overflow: hidden;"
                />
                <textarea
                  aria-hidden="true"
                  class="MuiOutlinedInput-input MuiInputBase-input MuiInputBase-inputMultiline css-1sqnrkk-MuiInputBase-input-MuiOutlinedInput-input"
                  readonly=""
                  style="visibility: hidden; position: absolute; overflow: hidden; height: 0px; top: 0px; left: 0px; transform: translateZ(0); padding: 0px; width: 100%;"
                  tabindex="-1"
                />
                <fieldset
                  aria-hidden="true"
                  class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                >
                  <legend
                    class="css-186xcr5"
                  >
                    <span>
                      app state
                    </span>
                  </legend>
                </fieldset>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
